fix: Add documentation (#1567)
embeddings-benchmark/mteb: CONTRIBUTING.md

## Contributing to mteb

We welcome contributions to `mteb` such as new tasks, code optimization or benchmarks.

Once you have decided on your contribution, this document describes how to set up the repository for development.

### Development Installation

If you want to submit a task or on other ways contribute to `mteb`, you will need to install the package in development mode:

```bash
# download the git repository
git clone https://github.com/embeddings-benchmark/mteb
cd mteb

# create your virtual environment and activate it
make install
```

> This uses [make](missing) to define the install command. You can see what each command does in the [makefile](missing).  

Reviewer: ```suggestion
This uses [make](https://www.gnu.org/software/make/) to define the install command. You can see what each command does in the [makefile](https://github.com/embeddings-benchmark/mteb/blob/main/Makefile).  
```

### Running Tests

To run the tests, you can use the following command:

```bash
make test
```

> This is also run by the CI pipeline, so if this passed locally you can be almost sure that your changes will not cause a failed test once you create a pull request. We recommend running the tests in the lowest version of python supported by the package (see the [pyproject.toml](missing)) to ensure compatibility.

Reviewer: ```suggestion
This is also run by the CI pipeline, so if this passed locally, you can be almost sure that your changes will not cause a failed test once you create a pull request. We recommend running the tests in the lowest version of python supported by the package (see the [pyproject.toml](https://github.com/embeddings-benchmark/mteb/blob/main/pyproject.toml)) to ensure compatibility.
```

### Running linting

> To run the linting before a pull request you can use the following command:

Reviewer: ```suggestion
To run the linting before submitting a pull request, use:
```
